otter/tap/api.py

```diff
@@ -309,18 +313,40 @@ def on_client_ready(_):
                 config_value('converger.step_limits') or {})
 
             # Setup selfheal service
-            shsvc = SelfHeal(
-                dispatcher, config_value("selfheal.interval") or 300,
-                log, reactor, config_value)
-            health_checker.checks["selfheal"] = shsvc.health_check
-            shsvc.setServiceParent(parent)
+            sh_svc = setup_selfheal_service(
+                reactor, config, dispatcher, health_checker, log)
+            parent.addService(sh_svc)
 
         d.addCallback(on_client_ready)
         d.addErrback(log.err, 'Could not start TxKazooClient')
 
     return parent
 
 
+def setup_selfheal_service(clock, config, dispatcher, health_checker, log):
+    """
+    Setup selfheal timer service and add it to ``parent``
```

Comment:
Good catch. I was passing `parent` earlier and adding this service to that but didn't feel comfortable with mutability aspect of the function. So moved the adding to caller but forgot to update the docs.